src/main/java/de/torui/coflsky/EventHandler.java

```diff
@@ -188,4 +202,165 @@ private static void ProcessScoreboard(String line){
+        return "";
+    }
+
+    public static void GuiEventHandler(GuiOpenEvent event) {
+        if (event.gui instanceof GuiContainer/* && GuiScreen.isShiftKeyDown()*/) {
+            new Thread(() -> {
+                try {
+                    Thread.sleep(50);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                InventoryWrapper wrapper = new InventoryWrapper();
+
+                GuiContainer gc = (GuiContainer) event.gui;
+
+                if (event.gui instanceof GuiChest) {
+                    ContainerChest chest = (ContainerChest) ((GuiChest) event.gui).inventorySlots;
+
+                    IInventory inv = chest.getLowerChestInventory();
+                    if (inv.hasCustomName()) {
+                        // verify that the chest actually has a custom name
+                        String chestName = inv.getName();
+                        wrapper.chestName = chestName;
+                    }
+                }
+
+                NBTTagCompound compound = new NBTTagCompound();
+                NBTTagList tl = new NBTTagList();
+
+                for (Slot obj : gc.inventorySlots.inventorySlots) {
+                    ItemStack stack = obj.getStack();
+                    if (stack != null) {
+                        tl.appendTag(stack.serializeNBT());
+                    } else {
+                        tl.appendTag(EMPTY_COMPOUND);
+                    }
+                }
+
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                try {
+                    compound.setTag("i", tl);
+                    CompressedStreamTools.writeCompressed(compound, baos);
+
+                    wrapper.fullInventoryNbt = Base64.getEncoder().encodeToString(baos.toByteArray());
+
+                    String data = WSClient.gson.toJson(wrapper);
+                    String info = QueryServerCommands.PostRequest("https://sky.coflnet.com/api/mod/description/modifications", data);
+
+                    DescModification[][] arr = WSClient.gson.fromJson(info, DescModification[][].class);
+                    int i = 0;
+                    for (Slot obj : gc.inventorySlots.inventorySlots) {
+                        ItemStack stack = obj.getStack();
+                        if (stack != null) {
+                            tooltipItemMap.put(stack, arr[i]);
+                            String uuid = ExtractUuidFromItemStack(stack);
+                            if(uuid.length()>0){
+                                tooltipItemUuidMap.put(uuid, arr[i]);
+                            }
+
+                            String id = ExtractStackableIdFromItemStack(stack);
+                            if(id.length()>0){
+                                tooltipItemIdMap.put(id, arr[i]);
+                            }
+
+                        }
+                        i++;
+                    }
+
+                } catch (IOException e) {
```

Comment:
please log exceptions.
Also this is pretty deeply nested, please consider refactoring the inner batch block into a separate function